src/Orleans.EventSourcing/Hosting/LogStorageSiloBuilderExtensions.cs

```diff
@@ -0,0 +1,35 @@
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Orleans.LogConsistency;
+using Orleans.Providers;
+using Orleans.Runtime;
+using Orleans.EventSourcing.LogStorage;
+
+namespace Orleans.Hosting
+{
+    public static class LogStorageSiloBuilderExtensions
+    {
+        /// <summary>
+        /// Adds a log storage log consistency provider as default consistency provider"/>
+        /// </summary>
+        public static ISiloHostBuilder AddLogStorageBasedLogConsistencyProviderAsDefault(this ISiloHostBuilder builder)
+        {
+            return builder.AddLogStorageBasedLogConsistencyProvider(ProviderConstants.DEFAULT_STORAGE_PROVIDER_NAME);
+        }
+
+        /// <summary>
+        /// Adds a log storage log consistency provider"/>
+        /// </summary>
+        public static ISiloHostBuilder AddLogStorageBasedLogConsistencyProvider(this ISiloHostBuilder builder, string name = "LogStorage")
+        {
+            return builder.ConfigureServices(services => services.AddLogStorageBasedLogConsistencyProvider(name));
+        }
+
+        internal static IServiceCollection AddLogStorageBasedLogConsistencyProvider(this IServiceCollection services, string name)
```

Comment:
what's the difference of `AddLogStorageBasedLogConsistencyProvider` and `AddCustomStorageBasedLogConsistencyProvider` ? seems to me they are all adding `ILogViewAdaptorFactory` , just the custom one add `CustomStorageLogConsistencyOptions` as well. the method name makes it sounds like to different LogConsistencyProviders. But is it ?

Reply:
I don't know the difference between these, but they are different. :/  It's not obvious because the way the code is organized.  All of these extension function groups add LogConsistencyProvider, but they are different due to namespace.  This one is using Orleans.EventSourcing.LogStorage.LogConsistencyProvider, while AddCustomStorageBasedLogConsistencyProvider adds Orleans.EventSourcing.CustomStorage.LogConsistencyProvider